flink-table-store-core/src/main/java/org/apache/flink/table/store/file/writer/AppendOnlyWriter.java

```diff
@@ -0,0 +1,173 @@
+import org.apache.flink.table.store.file.stats.FieldStats;
+import org.apache.flink.table.store.file.stats.FieldStatsCollector;
+import org.apache.flink.table.store.file.stats.FileStatsExtractor;
+import org.apache.flink.table.store.file.utils.FileUtils;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.util.Preconditions;
+
+import org.apache.flink.shaded.guava30.com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+
+/**
+ * A {@link RecordWriter} implementation that only accepts records which are always insert
+ * operations and don't have any unique keys or sort keys.
+ */
+public class AppendOnlyWriter implements RecordWriter {
+    private final BulkWriter.Factory<RowData> writerFactory;
+    private final RowType writeSchema;
+    private final long targetFileSize;
+    private final DataFilePathFactory pathFactory;
+    private final FileStatsExtractor fileStatsExtractor;
+
+    private final AtomicLong nextSeqNum;
+
+    private RowRollingWriter writer;
+
+    public AppendOnlyWriter(
+            FileFormat fileFormat,
+            long targetFileSize,
+            RowType writeSchema,
+            long maxWroteSeqNumber,
+            DataFilePathFactory pathFactory) {
+
+        this.writerFactory = fileFormat.createWriterFactory(writeSchema);
+        this.writeSchema = writeSchema;
+        this.targetFileSize = targetFileSize;
+
+        this.pathFactory = pathFactory;
+        this.fileStatsExtractor = fileFormat.createStatsExtractor(writeSchema).orElse(null);
+
+        this.nextSeqNum = new AtomicLong(maxWroteSeqNumber + 1);
+
+        this.writer = createRollingRowWriter();
+    }
+
+    @Override
+    public void write(ValueKind valueKind, RowData key, RowData value) throws Exception {
+        Preconditions.checkArgument(
+                valueKind == ValueKind.ADD,
+                "Append-only writer cannot accept ValueKind: " + valueKind);
+
+        writer.write(value);
+    }
+
+    @Override
+    public Increment prepareCommit() throws Exception {
+        List<DataFileMeta> newFiles = Lists.newArrayList();
+
+        if (writer != null) {
+            writer.close();
+
+            // Reopen the writer to accept further records.
+            newFiles.addAll(writer.result());
+            writer = createRollingRowWriter();
+        }
+
+        return new Increment(Lists.newArrayList(newFiles));
+    }
+
+    @Override
+    public void sync() throws Exception {
+        // Do nothing here, as this writer don't introduce any async compaction thread currently.
+    }
+
+    @Override
+    public List<DataFileMeta> close() throws Exception {
+        sync();
+
+        List<DataFileMeta> result = Lists.newArrayList();
+        if (writer != null) {
+            // Abort this writer to clear uncommitted files.
+            writer.abort();
```

Comment:
I'm just curious that the `RecordWriter#close` will close the writer and **delete the uncommitted files** automatically,  which is a bit mismatching the normal `close` semantic of writer.  [1]

Actually, the behavior is more matching the `abort` semantic. So I filed a separate issue to address this issue. I mean we can define separate `abort()` and `close()` methods in this interface,  and the `close()` will just close the opening writer, and the `abort()` will clean the uncommitted files. I think its semantic is more clear.

[1] https://github.com/apache/flink-table-store/blob/1fedb6586ba292bedf00bb1b6922d5110a2f7fa8/flink-table-store-core/src/main/java/org/apache/flink/table/store/file/writer/RecordWriter.java#L51

Reply:
Filed a separate issue for this: https://issues.apache.org/jira/browse/FLINK-27553

Reply:
Filed a separate issue for this: https://issues.apache.org/jira/browse/FLINK-27553

Reply:
Sounds good to me~